{
  "dataType": "CVE_RECORD",
  "containers": {
    "adp": [
      {
        "title": "CVE Program Container",
        "references": [
          {
            "url": "http://www.openwall.com/lists/oss-security/2024/08/22/1"
          },
          {
            "url": "http://www.openwall.com/lists/oss-security/2024/08/22/4"
          },
          {
            "url": "http://www.openwall.com/lists/oss-security/2024/08/23/1"
          },
          {
            "url": "http://www.openwall.com/lists/oss-security/2024/08/23/2"
          }
        ],
        "providerMetadata": {
          "orgId": "af854a3a-2127-422b-91ae-364da2661108",
          "shortName": "CVE",
          "dateUpdated": "2024-08-23T23:02:35.112Z"
        }
      },
      {
        "title": "CISA ADP Vulnrichment",
        "metrics": [
          {
            "other": {
              "type": "ssvc",
              "content": {
                "id": "CVE-2024-8088",
                "role": "CISA Coordinator",
                "options": [
                  {
                    "Exploitation": "none"
                  },
                  {
                    "Automatable": "yes"
                  },
                  {
                    "Technical Impact": "partial"
                  }
                ],
                "version": "2.0.3",
                "timestamp": "2024-08-23T17:18:49.209334Z"
              }
            }
          }
        ],
        "affected": [
          {
            "cpes": [
              "cpe:2.3:a:python_software_foundation:cpython:*:*:*:*:*:*:*:*"
            ],
            "vendor": "python_software_foundation",
            "product": "cpython",
            "versions": [
              {
                "status": "affected",
                "version": "0",
                "versionType": "custom",
                "lessThanOrEqual": "3.13.0"
              }
            ],
            "defaultStatus": "unknown"
          }
        ],
        "providerMetadata": {
          "orgId": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
          "shortName": "CISA-ADP",
          "dateUpdated": "2024-08-23T17:32:09.310Z"
        }
      }
    ],
    "cna": {
      "title": "Infinite loop when iterating over zip archive entry names from zipfile.Path",
      "source": {
        "discovery": "UNKNOWN"
      },
      "credits": [
        {
          "lang": "en",
          "type": "remediation developer",
          "value": "Jason R. Coombs"
        },
        {
          "lang": "en",
          "type": "coordinator",
          "value": "Seth Larson"
        }
      ],
      "metrics": [
        {
          "format": "CVSS",
          "cvssV4_0": {
            "Safety": "NEGLIGIBLE",
            "version": "4.0",
            "Recovery": "USER",
            "baseScore": 8.7,
            "Automatable": "NO",
            "attackVector": "NETWORK",
            "baseSeverity": "HIGH",
            "valueDensity": "NOT_DEFINED",
            "vectorString": "CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/S:N/AU:N/R:U/RE:L",
            "providerUrgency": "NOT_DEFINED",
            "userInteraction": "NONE",
            "attackComplexity": "LOW",
            "attackRequirements": "NONE",
            "privilegesRequired": "NONE",
            "subIntegrityImpact": "NONE",
            "vulnIntegrityImpact": "NONE",
            "subAvailabilityImpact": "NONE",
            "vulnAvailabilityImpact": "HIGH",
            "subConfidentialityImpact": "NONE",
            "vulnConfidentialityImpact": "NONE",
            "vulnerabilityResponseEffort": "LOW"
          },
          "scenarios": [
            {
              "lang": "en",
              "value": "GENERAL"
            }
          ]
        }
      ],
      "affected": [
        {
          "repo": "https://github.com/python/cpython",
          "vendor": "Python Software Foundation",
          "product": "CPython",
          "versions": [
            {
              "status": "affected",
              "version": "0",
              "versionType": "python",
              "lessThanOrEqual": "3.13.0"
            }
          ],
          "defaultStatus": "unaffected"
        }
      ],
      "references": [
        {
          "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/GNFCKVI4TCATKQLALJ5SN4L4CSPSMILU/",
          "tags": [
            "vendor-advisory"
          ]
        },
        {
          "url": "https://github.com/python/cpython/pull/122906",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/issues/122905",
          "tags": [
            "issue-tracking"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/795f2597a4be988e2bb19b69ff9958e981cb894e",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/8c7348939d8a3ecd79d630075f6be1b0c5b41f64",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/dcc5182f27c1500006a1ef78e10613bb45788dea",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/e0264a61119d551658d9445af38323ba94fc16db",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/issues/123270",
          "tags": [
            "issue-tracking"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/2231286d78d328c2f575e0b05b16fe447d1656d6",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/7e8883a3f04d308302361aeffc73e0e9837f19d4",
          "tags": [
            "patch"
          ]
        },
        {
          "url": "https://github.com/python/cpython/commit/95b073bddefa6243effa08e131e297c0383e7f6a",
          "tags": [
            "patch"
          ]
        }
      ],
      "x_generator": {
        "engine": "Vulnogram 0.2.0"
      },
      "descriptions": [
        {
          "lang": "en",
          "value": "There is a HIGH severity vulnerability affecting the CPython \"zipfile\"\nmodule affecting \"zipfile.Path\". Note that the more common API \"zipfile.ZipFile\" class is unaffected.\n\n\n\n\n\nWhen iterating over names of entries in a zip archive (for example, methods\nof \"zipfile.Path\" like \"namelist()\", \"iterdir()\", etc)\nthe process can be put into an infinite loop with a maliciously crafted\nzip archive. This defect applies when reading only metadata or extracting\nthe contents of the zip archive. Programs that are not handling\nuser-controlled zip archives are not affected.",
          "supportingMedia": [
            {
              "type": "text/html",
              "value": "<div>There is a HIGH severity vulnerability affecting the CPython \"zipfile\"\nmodule affecting \"zipfile.Path\". Note that the more common API \"zipfile.ZipFile\" class is unaffected.<br></div><div><br></div><div>When iterating over names of entries in a zip archive (for example, methods\nof \"zipfile.Path\" like \"namelist()\", \"iterdir()\", etc)\nthe process can be put into an infinite loop with a maliciously crafted\nzip archive. This defect applies when reading only metadata or extracting\nthe contents of the zip archive. Programs that are not handling\nuser-controlled zip archives are not affected.</div>",
              "base64": false
            }
          ]
        }
      ],
      "problemTypes": [
        {
          "descriptions": [
            {
              "lang": "en",
              "type": "CWE",
              "cweId": "CWE-835",
              "description": "CWE-835 Loop with Unreachable Exit Condition ('Infinite Loop')"
            }
          ]
        }
      ],
      "providerMetadata": {
        "orgId": "28c92f92-d60d-412d-b760-e73465c3df22",
        "shortName": "PSF",
        "dateUpdated": "2024-09-03T15:12:33.775Z"
      }
    }
  },
  "cveMetadata": {
    "cveId": "CVE-2024-8088",
    "state": "PUBLISHED",
    "dateUpdated": "2024-09-03T15:12:33.775Z",
    "dateReserved": "2024-08-22T12:42:32.661Z",
    "assignerOrgId": "28c92f92-d60d-412d-b760-e73465c3df22",
    "datePublished": "2024-08-22T18:45:31.807Z",
    "assignerShortName": "PSF"
  },
  "dataVersion": "5.1"
}